plaso/parsers/gdrive_synclog.py

```diff
@@ -0,0 +1,197 @@
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import manager
+from plaso.parsers import text_parser
+
+
+class GoogleDriveSyncLogEventData(events.EventData):
+  """Google Drive Sync log event data.
+
+  Attributes:
+    log_level (str): logging level of event such as "DEBUG", "WARN", "INFO",
+        "ERROR".
+    message (str): log message.
+    pid (int): process ID of process which logged event.
+    source_code (str): filename:line_number of source file which logged event.
+    thread (str): colon-separated thread identifier in the form "ID:name"
+        which logged event.
+    time (str): date and time of the log entry event with timezone offset.
+  """
+
+  DATA_TYPE = 'gdrive_sync:log:line'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(GoogleDriveSyncLogEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.time = None
+    self.log_level = None
+    self.pid = None
+    self.thread = None
+    self.source_code = None
+    self.message = None
+
+
+class GoogleDriveSyncLogParser(text_parser.PyparsingMultiLineTextParser):
+  """Parses events from Google Drive Sync log files."""
+
+  NAME = 'gdrive_synclog'
+
+  DESCRIPTION = 'Parser for Google Drive Sync log files.'
+
+  _ENCODING = 'utf-8'
+
+  # Increase the buffer size, as log messages are often many lines of python
+  # object dumps or similar.
+  BUFFER_SIZE = 16384
+
+  # Common structures.
+  _HYPHEN = text_parser.PyparsingConstants.HYPHEN
+
+  _FOUR_DIGITS = text_parser.PyparsingConstants.FOUR_DIGITS
+  _THREE_DIGITS = text_parser.PyparsingConstants.THREE_DIGITS
+  _TWO_DIGITS = text_parser.PyparsingConstants.TWO_DIGITS
+
+  # Ignore commas and spaces.
+  IGNORE_FIELD = pyparsing.CharsNotIn(',').suppress()
+
+  _GDS_DATE_TIME = (
+      _FOUR_DIGITS.setResultsName('year') + _HYPHEN +
+      _TWO_DIGITS.setResultsName('month') + _HYPHEN +
+      _TWO_DIGITS.setResultsName('day') +
+      text_parser.PyparsingConstants.TIME_MSEC_ELEMENTS +
+      pyparsing.Word(pyparsing.printables).setResultsName('time_zone_offset')
+  ).setResultsName('date_time')
+
+  # Multiline entry end marker, matched from right to left.
+  _GDS_ENTRY_END = pyparsing.StringEnd() | _GDS_DATE_TIME
+
+  _GDS_LINE = (
+      _GDS_DATE_TIME +
+      pyparsing.Word(pyparsing.alphas).setResultsName('log_level') +
+      # TODO: strip pid= out, cast to integers?
+      pyparsing.Word(pyparsing.printables).setResultsName('pid') +
+      # TODO: consider stripping thread identifier/cleaning up thread name?
+      pyparsing.Word(pyparsing.printables).setResultsName('thread') +
+      pyparsing.Word(pyparsing.printables).setResultsName('source_code') +
+      pyparsing.SkipTo(_GDS_ENTRY_END).setResultsName('message') +
+      pyparsing.ZeroOrMore(pyparsing.lineEnd()))
+
+  LINE_STRUCTURES = [
+      ('logline', _GDS_LINE),
+  ]
+
+  def _ParseLine(self, parser_mediator, structure):
```

Comment:
Name and docstring of the function do not represent the actual logic. This method can parse more than 1 line.

Maybe rename to: _ParseRecordLogline

Reply:
Done.